src/XMakeTasks/Copy.cs

```diff
@@ -408,6 +409,20 @@ CopyFileWithState copyFile
                 _copiedFiles = new TaskItem[0];
                 return true;
             }
+            
+            if (_destinationFiles != null)
+            {
+                var sameDestinationFiles = _destinationFiles.GroupBy(x => x).Where(g => g.Count() > 1).Select(y => y.First());
```

Comment:
I'm not so sure selecting the first of every set of duplicates is the best way to go. I think I'd prefer an error message more like "Cannot copy multiple files to the same destination. SourceFiles 'X', 'Y', and 'Z' all have the destination 'Q'."